pkg/controller/elasticsearch/driver/downscale_invariants.go

```diff
@@ -16,30 +16,37 @@ import (
 const (
 	OneMasterAtATimeInvariant        = "A master node is already in the process of being removed"
 	AtLeastOneRunningMasterInvariant = "Cannot remove the last running master node"
+	RespectMaxUnavailableInvariant   = "Not removing node to respect maxUnavailable setting"
 )
 
 // checkDownscaleInvariants returns true if the given state state allows downscaling the given StatefulSet.
 // If not, it also returns the reason why.
 func checkDownscaleInvariants(state downscaleState, statefulSet appsv1.StatefulSet) (bool, string) {
-	if !label.IsMasterNodeSet(statefulSet) {
-		// only care about master nodes
-		return true, ""
+	if state.removalsAllowed != nil && *state.removalsAllowed < 1 {
+		// removalsAllowed is nil then removals shouldn't be bounded here
```

Comment:
`removalsAllowed` is **not** nil here?

Reply:
I think I meant it to begin with 'if' :) added it and moved up.